### PR TITLE
Make deployments consistently succeed  

### PR DESCRIPTION
**Pull request by karlfloersch**
Currently deployments fail often, this PR makes them slower but more consistent.

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol [hunk 1]
```diff
@@ -355,7 +355,7 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
 
         for (uint32 i = 0; i < numContexts; i++) {
             BatchContext memory context = _getBatchContext(i);
-            _validateBatchContext(context, nextQueueIndex);
+            // _validateBatchContext(context, nextQueueIndex);
```
Comment (tynes) [inline contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol:358]: Did you mean to comment this out?

File: src/contract-deployment/deploy.ts [hunk 2]
```diff
@@ -16,15 +16,18 @@ export interface DeployResult {
 export const deploy = async (
   config: RollupDeployConfig
 ): Promise<DeployResult> => {
+  console.log('Beginning deploy')
```
Comment (tynes) [inline src/contract-deployment/deploy.ts:19]: These logs will break the `deployer` container

File: src/contract-deployment/config.ts [hunk 3]
```diff
@@ -65,11 +65,12 @@ export const makeContractDeployConfig = async (
         )
           .connect(config.deploymentSigner)
           .attach(contracts.Proxy__OVM_L1CrossDomainMessenger.address)
-        await xDomainMessenger.initialize(AddressManager.address)
-        await AddressManager.setAddress(
+        await (await xDomainMessenger.initialize(AddressManager.address, config.deployOverrides)).wait()
```
Comment (tynes) [inline src/contract-deployment/config.ts:68]: Not a huge fan of nested awaits on the same line, it feels pretty busy

**Comment by tynes** [comment]
See this approach for a faster approach: https://github.com/ethereum-optimism/contracts-v2/pull/51/files#diff-3d2aa7498c1a7bdecb7dfd7b4b32db4b9002a9dd0093316455609da76ff952f1R60

**Comment by karlfloersch** [comment]
> See this approach for a faster approach: https://github.com/ethereum-optimism/contracts-v2/pull/51/files#diff-3d2aa7498c1a7bdecb7dfd7b4b32db4b9002a9dd0093316455609da76ff952f1R60

Unfortunately on Goerli sending more than 1 transaction without waiting for the receipt often causes the deployment to fail :( So that should work if mempool management was better but unfortunately we've had issues in the past.

**Comment by karlfloersch** [comment]
Closing in favor of https://github.com/ethereum-optimism/contracts-v2/pull/99

**Pull request closed without merging**